### PR TITLE
PS-1006 Return empty string instead of nil

### PR DESCRIPTION
**Pull request by codingrhythm**
Currently if it fails to convert the `std::string` into `NSString*`, it returns `nil`. This can be a problem for `NSDictionary` as it does not allow `nil` to be the value, thus will have runtime crash when following line is called.
 https://github.com/SafetyCulture/djinni/blob/6de462b3ef701a9a2103c8ee269c575fb6d96524/support-lib/objc/DJIMarshal%2BPrivate.h#L348

According to https://developer.apple.com/documentation/foundation/nsstring/1407339-initwithbytes?language=objc

```Objective-C
[[NSString alloc] initWithBytes:string.data()
                         length:static_cast<NSUInteger>(string.size())
                       encoding:NSUTF8StringEncoding];
```

> If the length of the byte string is greater than the specified length a nil value is returned.

In order to prevent it from crashing, we return an empty string instead of `nil`.  Since we already lose the value if a `nil` is returned, this change should not make things worse.

Please note, this is not the fix for the [crash](https://console.firebase.google.com/u/0/project/project-8638960581077738091/crashlytics/app/ios:com.safetyculture.iAuditor/issues/b2c7da2e3ec541f60c5299913d3fc657?time=last-thirty-days&sessionId=349b4dde8f1048bd9aabe29f85e75f63_DNE_0_v2), I tried multiple options, but only this change can give me high confidence. I will explain more in following Crux PR. Also, feel free to raise any concerns.

### CODE REVIEW

**Review by rogchap — APPROVED** [review]
Makes sense.

**Comment by jimmyhoran** [comment]
Do we know when and why it fails to convert `std::string` into `NSString` 🤔

**Comment by jimmyhoran** [comment]
> Do we know when and why it fails to convert `std::string` into `NSString` 🤔

Argh, sorry I miss read the Apple link reference.
Ignore my message above ☝️

**Review by jimmyhoran — APPROVED** [review]

File: support-lib/objc/DJIMarshal+Private.h [hunk 1]
```diff
@@ -122,9 +122,14 @@ struct String {
...
-                                        length:static_cast<NSUInteger>(string.size())
-                                      encoding:NSUTF8StringEncoding];
+        ObjcType result = [[NSString alloc] initWithBytes:string.data()
+                                                   length:static_cast<NSUInteger>(string.size())
+                                                 encoding:NSUTF8StringEncoding];
+        if (!result) {
+            return @"";
+        }
+
+        return result;
```
Comment (jimmyhoran) [inline support-lib/objc/DJIMarshal+Private.h:134]: Might be good to add a comment here to detail what this solves 👍

**Review by jimmyhoran — APPROVED** [review]

**Comment by codingrhythm** [comment]
> Do we know when and why it fails to convert `std::string` into `NSString` 🤔

I will have more details in Crux PR that tries to explain why we may end up a string like that.

**Pull request merged**